File: 백준/Silver/15664. N과 M （10）/N과 M （10）.js
const filePath = process.platform === 'linux' ? '/dev/stdin' : 'e.txt';
const input = require('fs').readFileSync(filePath).toString().trim().split('\n');

/*
왼쪽 상단에서 좌표를 시작한다. 다만 좌표는 1, 1부터 시작한다. 
*/

const [count, maxLength] = input[0].split(' ').map(Number);
const numbers = input[1]
  .split(' ')
  .map(Number)
  .sort((a, b) => a - b);

const temp = [];
const resultList = [];
function bt(idx) {
  if (temp.length === maxLength) {
    resultList.push(temp.join(' '));
    return;
  }

  let value = null;
  for (let i = idx; i < count; i++) {
    if (value === numbers[i]) continue;
    value = numbers[i];

    temp.push(numbers[i]);
    bt(i + 1);
    temp.pop();
  }
}

bt(0);
console.log(resultList.join('\n'));

File: 백준/Silver/15664. N과 M （10）/N과 M （10）.py
n, m = map(int, input().split())
lst = list(map(int, input().split()))

lst.sort()

temp = []
def comb(curIdx):
    if len(temp) == m:
        print(' '.join(map(str, temp)))
        return
    
    prev = None
    for i in range(curIdx, n):
        cur = lst[i]
        
        if cur == prev:
            continue
        
        prev = cur
        
        temp.append(cur)
        comb(i + 1)
        temp.pop()
    
comb(0)